pkg/network/setup/netpod/netpod_test.go

```diff
@@ -907,20 +936,21 @@ var _ = Describe("netpod", func() {
 				{
 					Name:          "somenet",
 					NetworkSource: v1.NetworkSource{Pod: &v1.PodNetwork{}},
-					//NetworkSource: v1.NetworkSource{Multus: &v1.MultusNetwork{NetworkName: "somenad"}},
 				},
 			},
 			[]v1.Interface{{Name: "somenet", InterfaceBindingMethod: binding}},
-			vmiUID, 0, 0, 0,
+			vmiUID, 0, 0, 0, state,
 			netpod.WithNMStateAdapter(&nmstatestub),
 			netpod.WithCacheCreator(&baseCacheCreator),
 		)
-		Expect(netPod.Setup(noopPostDiscoveryHook)).To(Succeed())
-		Expect(nmstatestub.spec).To(Equal(nmstate.Spec{Interfaces: []nmstate.Interface{}}))
+		Expect(netPod.Setup()).To(Succeed())
+		Expect(nmstatestub.spec).To(Equal(expNmstateSpec))
 	},
-		Entry("SR-IOV", v1.InterfaceBindingMethod{SRIOV: &v1.InterfaceSRIOV{}}),
-		Entry("Slirp", v1.InterfaceBindingMethod{Slirp: &v1.InterfaceSlirp{}}),
-		Entry("Macvtap", v1.InterfaceBindingMethod{Macvtap: &v1.InterfaceMacvtap{}}),
+		// Not processed by the discovery & config steps.
+		Entry("SR-IOV", v1.InterfaceBindingMethod{SRIOV: &v1.InterfaceSRIOV{}}, nmstate.Spec{}),
+		Entry("Macvtap", v1.InterfaceBindingMethod{Macvtap: &v1.InterfaceMacvtap{}}, nmstate.Spec{}),
+		// Processed by the discovery but not by the config step.
+		Entry("Slirp", v1.InterfaceBindingMethod{Slirp: &v1.InterfaceSlirp{}}, nmstate.Spec{Interfaces: []nmstate.Interface{}}),
```

Comment:
Can you please explain why with no discovery the `nmstate.Spec` has `nil` as `Interfaces`. Why with discovery the `Interfaces` are an empty list? Shouldn't in both case the `Interfaces` be untouched (`nil`)?

Reply:
I find it confusing for a future developer that will have to understand this test.
Please at least add a comment about it.

Reply:
I do not like to add this story, it feels too low level detail and may potentially rot if things change.

Nevertheless, it also does not deserve too much investment in discussing it.
Done.